src/rust/engine/src/scheduler.rs

```diff
@@ -44,6 +44,24 @@ type ObservedValueResult = Result<(Value, Option<LastObserved>), Failure>;
 // Root requests are limited to Select nodes, which produce (python) Values.
 type Root = Select;
 
+// When enabled, the interval at which all stragglers that have been running for longer than a
+// threshold should be logged. The threshold might become configurable, but this might not need
+// to be.
+const STRAGGLER_INTERVAL: Duration = Duration::from_secs(30);
+
+///
+/// An enum for the two cases of `--[no-]dynamic-ui`.
+///
+enum SessionDisplay {
+  // The dynamic UI is enabled, and the ConsoleUI should interact with a TTY.
+  ConsoleUI(ConsoleUI),
+  // The dynamic UI is disabled, and we should use only logging.
+  Logging {
+    straggler_threshold: Duration,
+    straggler_deadline: Option<Instant>,
```

Comment:
Maybe call this `straggler_logging_deadline`?

Reply:
It's in a struct called `Logging`, so I think that we can skip that.